File: Témata/04 nelineární regresní model (M-H algoritmus)/Matlab/gamrnd_Koop.m
function y = gamrnd_Koop(mu,nu,M,N)
%Generovani nahodnych cisel z G(mu,nu)
%dle Koop(2003), vyuzivajici gammrnd
%statistickeho toolboxu
%M ... pocet radku
%N ... pocet sloupcu

A = nu/2;
B = 2*mu/nu;
y = gamrnd(A,B,M,N);
end

File: Témata/01_NLRM_s_prirozene_konjugovanou_apriorni_hustotou/Python/Support/gamrnd_Koop.py
from numpy.random import gamma
from numpy import reshape

'''
Funkce pro generovani nahodnych cisel z G(mu, nu)
- podle Koop(2003)

    M ... pocet radku
    N ... pocet sloupcu
'''

def gamrnd_Koop(mu, nu, M, N):
    A = nu/2
    B = 2 * mu/nu

    # tato funkce vraci vektor o delce M x N
    y = gamma(A, B, M*N)

    Y = reshape(y,(M,N))

    return Y
